server/src/main/java/org/apache/iotdb/db/metrics/server/ServerArgument.java

```diff
@@ -0,0 +1,347 @@
+			e.printStackTrace();
+		}
+		return ia.getHostName();
+	}
+
+	private String osName() {
+		return System.getProperty("os.name");
+	}
+
+	private int totalCores() {
+		OperatingSystemMXBean osmxb = (OperatingSystemMXBean) ManagementFactory.getOperatingSystemMXBean();
+		int freeCores = osmxb.getAvailableProcessors();
+		return freeCores;
+	}
+
+	private long totalMemory() {
+		return Runtime.getRuntime().totalMemory() / 1024 / 1024;
+	}
+
+	private long freeMemory() {
+		return Runtime.getRuntime().freeMemory() / 1024 / 1024;
+	}
+
+	private long maxMemory() {
+		return Runtime.getRuntime().maxMemory() / 1024 / 1024;
+	}
+
+	private long totalPhysicalMemory() {
+		OperatingSystemMXBean osmxb = (OperatingSystemMXBean) ManagementFactory.getOperatingSystemMXBean();
+		long totalMemorySize = osmxb.getTotalPhysicalMemorySize() / 1024 / 1024;
+		return totalMemorySize;
+	}
+
+	private long usedPhysicalMemory() {
+		OperatingSystemMXBean osmxb = (OperatingSystemMXBean) ManagementFactory.getOperatingSystemMXBean();
+		long usedMemorySize = (osmxb.getTotalPhysicalMemorySize() - osmxb.getFreePhysicalMemorySize()) / 1024 / 1024;
+		return usedMemorySize;
+	}
+
+	private long freePhysicalMemory() {
+		OperatingSystemMXBean osmxb = (OperatingSystemMXBean) ManagementFactory.getOperatingSystemMXBean();
+		long freeMemorySize = osmxb.getFreePhysicalMemorySize() / 1024 / 1024;
+		return freeMemorySize;
+	}
+
+	public int getPort() {
+		return port;
+	}
+
+	public void setPort(int port) {
+		this.port = port;
+	}
+
+	public String getHost() {
+		return host;
+	}
+
+	public int getCores() {
+		return cores;
+	}
+
+	public long getTotalMemory() {
+		return totalMemory;
+	}
+
+	public long getFreeMemory() {
+		return freeMemory;
+	}
+
+	public long getMaxMemory() {
+		return maxMemory;
+	}
+
+	public String getOsName() {
+		return osName;
+	}
+
+	public long getTotalPhysicalMemory() {
+		return totalPhysicalMemory;
+	}
+
+	public long getFreePhysicalMemory() {
+		return freePhysicalMemory;
+	}
+
+	public long getUsedPhysicalMemory() {
+		return usedPhysicalMemory;
+	}
+
+	public double getCpuRatio() {
+		String osName = System.getProperty("os.name").toLowerCase();
+		cpuRatio = 0;
+		if (osName.indexOf("windows") >= 0) {
+			cpuRatio = getCpuRatioForWindows();
+		} else if(osName.indexOf("linux") >= 0){
+			cpuRatio = getCpuRateForLinux();
+		} else {
+			try {
+				throw new Exception("Can't get the cpu ratio,because this OS is not support");
+			} catch (Exception e) {
+				e.printStackTrace();
+			}
+		}
+		return cpuRatio;
+	}
+
+	/**
+	 * cpu ratio for linux
+	 */
+	private double getCpuRateForLinux() {
+		try {
+			long[] c0 = readLinuxCpu();
+			Thread.sleep(CPUTIME);
+			long[] c1 = readLinuxCpu();
+			if (c0 != null && c1 != null) {
+				long idleCpuTime = c1[0] - c0[0];
+				long totalCpuTime = c1[1] - c0[1];
+				if (totalCpuTime == 0) {
+					return 100.0;
+				}
+				return Double.valueOf(100 * (1 - idleCpuTime / totalCpuTime)).doubleValue();

```

Comment:
Devision of these two long value `idleCpuTime` and `totalCpuTime` is either 0 or 1. If you want to get double result, you can try to first parse long values to double values.